src/components/Dropdown.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { Fragment } from "react";
+import { Menu, Transition } from "@headlessui/react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import { atom, useAtom } from "jotai";
+
+// —————————————————————————————————————————————————————————————————————————————
+// Environment
+
+function classNames(...classes: string[]) {
+  return classes.filter(Boolean).join(" ");
+}
+
+type DropdownProps<T> = React.PropsWithChildren<{
+  className?: string;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  name: string;
+  default: T;
+}>;
+
+type DropItem = React.PropsWithChildren<{
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+}>;
+
+const selectedAtom = atom("");
+
+// —————————————————————————————————————————————————————————————————————————————
+// Component
+
+export function DropItem<T>({ children, onClick, ...props }: DropItem) {
+  const [, setSelected] = useAtom(selectedAtom);
```

Comment:
We talked about this, setSelected is used like a global variable/function. We can do it this way.